tsl/src/continuous_aggs/materialize.c

```diff
@@ -392,78 +374,29 @@ get_materialization_end_point_for_table(int32 raw_hypertable_id, int32 materiali
 			 end_time);
 
 	Assert(end_time > old_completed_threshold);
-	Assert(end_time >= start_time);
 
 	*materializing_new_range = true;
 	return end_time;
 }
 
-static bool
-hypertable_get_min_and_max(SchemaAndName hypertable, Name time_column, int64 search_start,
-						   Oid time_type, int64 *min_out, int64 *max_out)
+static int64
+hypertable_get_min(SchemaAndName hypertable, Name time_column, Oid time_type)
 {
-	Datum last_time_value;
-	Datum first_time_value;
+	Datum min_time_datum;
 	bool val_is_null;
-	bool search_start_is_infinite = false;
-	bool found_new_tuples = false;
 	StringInfo command = makeStringInfo();
 	int res;
-
-	Datum search_start_val =
-		internal_to_time_value_or_infinite(search_start, time_type, &search_start_is_infinite);
-
-	if (search_start_is_infinite && search_start > 0)
-	{
-		/* the previous completed time was +infinity, there can be no new ranges */
-		return false;
-	}
+	int64 min_time_internal = PG_INT64_MIN;
 
 	res = SPI_connect();
 	if (res != SPI_OK_CONNECT)
 		elog(ERROR, "could not connect to SPI while search for new tuples");
 
-	/* We always SELECT both max and min in the following queries. There are two cases
-	 * 1. there is no index on time: then we want to perform only one seqscan.
-	 * 2. there is a btree index on time: then postgres will transform the query
-	 *    into two index-only scans, which should add very little extra work
-	 *    compared to the materialization.
-	 * Ordered append append also fires, so we never scan beyond the first and last chunks
-	 */
-	if (search_start_is_infinite)
-	{
-		/* previous completed time is -infinity, or does not exist, so we must scan from the
-		 * beginning */
-		appendStringInfo(command,
-						 "SELECT max(%s), min(%s) FROM %s.%s",
-						 quote_identifier(NameStr(*time_column)),
-						 quote_identifier(NameStr(*time_column)),
-						 quote_identifier(NameStr(*hypertable.schema)),
-						 quote_identifier(NameStr(*hypertable.name)));
-	}
-	else
-	{
-		Oid out_fn;
-		bool type_is_varlena;
-		char *search_start_str;
-
-		getTypeOutputInfo(time_type, &out_fn, &type_is_varlena);
-
-		search_start_str = OidOutputFunctionCall(out_fn, search_start_val);
-
-		*min_out = search_start;
-		/* normal case, add a WHERE to take advantage of chunk constraints */
-		/*We handled the +infinity case above*/
-		Assert(!search_start_is_infinite);
-		appendStringInfo(command,
-						 "SELECT max(%s), min(%s) FROM %s.%s WHERE %s >= %s",
-						 quote_identifier(NameStr(*time_column)),
-						 quote_identifier(NameStr(*time_column)),
-						 quote_identifier(NameStr(*hypertable.schema)),
-						 quote_identifier(NameStr(*hypertable.name)),
-						 quote_identifier(NameStr(*time_column)),
-						 quote_literal_cstr(search_start_str));
-	}
+	appendStringInfo(command,
+					 "SELECT min(%s) FROM %s.%s",
```

Comment:
how come the removal of `WHERE time >= old_completed_threshold`?

Reply:
this is only called if old_completed_threshold is not set